### PR TITLE
Catalogue of my things

### PR DESCRIPTION
**Pull request by ishimwezachee**
## In this project, we have implemented the following;
- Insert and query data in SQL.
- Use primary key & foreign key mechanisms for joining tables.
- Understand the different types of relationships between tables.
- Query multiple tables.
- Prepare complex queries that answer analytical questions.
- Run a program using the command line.
- Use Ruby syntax for basic programming operations.
- Apply Ruby best practices and language style guides in code.
- Describe the SOLID principles of OOP.
- Implement classes and objects in Ruby.
- Understand the four main principles of OOP.
- Implement encapsulation and inheritance with Ruby.
- Create a UML class diagram.
- Explain the difference between associations, aggregations, and composition in OOP.
- Set up associations between classes and objects.
- Store data in files.
- Build interactive console apps.
- Recognize the value of making equal contributions to group projects to produce the best outcome.

### CODE REVIEW

File: modules/label_module.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+require 'date'
+
+module LabelsController
+  def load_labels
+    file = './json_files/labels.json'
+    data = []
+    if File.exist?(file) && File.read(file) != ''
+      JSON.parse(File.read(file)).each do |element|
+        data.push(Label.new(id: element['id'].to_i, title: element['title'], color: element['color']))
+      end
```
Comment (Dipeshtwis) [inline modules/label_module.rb:12]: - [ ] Labels are saved in the JSON file but when I tried to fetch, it says "There are no labels yet!". Please fix this.
As you can see in the screenshot below. We have labels stored but when tried to fetch, it says no label.

Screenshot from 2022-02-25 21-18-59

File: spec/genre_spec.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+describe Genre do
+  context 'It should create a Genre instance with name' do
+    genre = Genre.new('Way to the moon')
+    it 'creates new Genre with the given parameters' do
+      expect(genre).to be_an_instance_of Genre
+    end
+    #   it 'should check if it can be archived' do
+    #     expect(genre.can_be_archived?).to eq true
+    #   end
```
Comment (Dipeshtwis) [inline spec/genre_spec.rb:11]: - [ ] I would recommend removing the commented line to make the codebase cleaner.

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @Team, 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.
:+1: I would appreciate the hard work you have done by implementing the **source** class as well as it was only required when you were 4 people. :clap: :clap: 
:+1: All tests are passing. :heavy_check_mark: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


 ## Required Changes ♻️

**_Check the comments under the review_**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- N.A.

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**_Dipesh Kumar_**


------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Lameck1 — APPROVED** [review]
Hi, @RolandM99, @aakbarkhan, and @ishimwezachee,

Tasks for this project have been successfully implemented 💯.
You've done well working on the concerns from the previous review 👏🏿.

# Highlights
- Professional README ✔️
- Clean code ✔️ 
- Good commit messages ✔️ 
- The MVP working as expected ✔️ 

All the best with the next project.

You are free to merge this branch :shipit:
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Please, remember to tag me in your question so I can receive the notification.**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**